File: TwoSum/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
let twoSum = function(nums, target) {
    let res = {};
    
    let num;
    for (let i = 0; i < nums.length; i++) {
        num = nums[i]
        if (target - num in res) {
            return [res[target - num], i]
        }
        res[num] = i
    }
};

File: TwoSum/two_sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = hashMapOf<Int, Int>()

        nums.forEachIndexed { i, num ->
            if (target - num in map) {
                return intArrayOf(map[target - num]!!, i)
            }
            map.put(num, i)
        }
        throw IllegalArgumentException("No two sum solution")
    }
}
